### ISSUE TITLE
 Private method wrongly marked as "unused" when called from sibling

### ISSUE THREAD

**Issue by KingCrunch** [phpmd/phpmd#252]
```php
<?php
class Foo
{
    public static function staticMethod($bar)
    {
        $x = new self;

        return $x->privateMethod($bar);
    }

    private function privateMethod($bar)
    {
        return $bar;
    }
}

$a = new Foo;
echo $a->staticMethod('Hello world');
```
See http://3v4l.org/G9sVq

`phpmd` markes the method `privateMethod()` as "unused".

**Comment by yoosefi** [comment]
Try tagging the static method with `@uses self::privateMethod()`

**Comment by manuelpichler** [comment]
I would expect this may work in a far far away future, because we don't have data flow analysis implemented.

**Comment by chriscalip** [comment]
Error sighted; on my case ```Avoid unused private fields such as``` but class has traits with methods that makes used of said fields.
